test/opt/test_eliminate_dead_const.cpp

```diff
@@ -0,0 +1,759 @@
+      .AppendInMain(tc.main_insts);
+  std::string expected = builder.GetCode();
+  builder.AppendTypesConstantsGlobals(tc.dead_consts);
+  std::string assembly_with_dead_const = builder.GetCode();
+  SinglePassRunAndCheck<opt::EliminateDeadConstantPass>(
+      assembly_with_dead_const, expected, /*  skip_nop = */ true);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    ScalarTypeConstants, EliminateDeadConstantTest,
+    ::testing::ValuesIn(std::vector<EliminateDeadConstantTestCase>({
+        // Scalar type constants, one dead constant and one used constant.
+        {
+            /* .used_consts = */
+            {
+                "%used_const_int = OpConstant %int 1",
+            },
+            /*  .main_insts = */
+            {
+                "%int_var = OpVariable %_pf_int Function",
+                "OpStore %int_var %used_const_int",
+            },
+            /* .dead_consts = */
+            {
+                "%dead_const_int = OpConstant %int 1",
+            },
+        },
+        {
+            /* .used_consts = */
+            {
+                "%used_const_uint = OpConstant %uint 1",
+            },
+            /*  .main_insts = */
+            {
+                "%uint_var = OpVariable %_pf_uint Function",
+                "OpStore %uint_var %used_const_uint",
+            },
+            /* .dead_consts = */
+            {
+                "%dead_const_uint = OpConstant %uint 1",
+            },
+        },
+        {
+            /* .used_consts = */
+            {
+                "%used_const_float = OpConstant %float 3.14",
+            },
+            /*  .main_insts = */
+            {
+                "%float_var = OpVariable %_pf_float Function",
+                "OpStore %float_var %used_const_float",
+            },
+            /* .dead_consts = */
+            {
+                "%dead_const_float = OpConstant %float 3.14",
+            },
+        },
+        {
+            /* .used_consts = */
+            {
+                "%used_const_double = OpConstant %double 3.14",
+            },
+            /*  .main_insts = */
+            {
+                "%double_var = OpVariable %_pf_double Function",
+                "OpStore %double_var %used_const_double",
+            },
+            /* .dead_consts = */
+            {
+                "%dead_const_double = OpConstant %double "
+                "3.14",
+            },
+        },
+    })));
+
+INSTANTIATE_TEST_CASE_P(
+    VectorTypeConstants, EliminateDeadConstantTest,
+    ::testing::ValuesIn(std::vector<EliminateDeadConstantTestCase>({
+        // Tests eliminating dead constant type ivec2. One dead constant vector
+        // and one used constant vector, each built from its own group of
+        // scalar constants.
+        {
+            /* .used_consts = */
+            {
+                "%used_int_x = OpConstant %int 1",
+                "%used_int_y = OpConstant %int 2",
+                "%used_v2int = OpConstantComposite %v2int "
```

Comment:
Since we use manual format for this kind of list, it would be nice to avoid splitting the same string into multiple lines. I feel it's quite hard to read and differentiate between multiple strings (via `,`) or the same string crossing multiple lines (without `,`).